pkg/trace/agent/test_utils.go

```diff
@@ -0,0 +1,19 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package agent
+
+import (
+	"context"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/config"
+)
+
+func agentWithDefaults() (agnt *Agent, stop func()) {
```

Comment:
nit: I'd personally prefer avoiding test utils files. Anecdotally, I don't think they are terribly common as a best practice. I realize that you're doing that now because the unit tests and fuzz tests have to live in separate files due to the build constraint, but eventually that's something we should clean up once <1.18 is no longer supported (i.e. at that point, just move the fuzz tests and unit tests into the same file). For now, it seems fine to me to leave any of these common functions in the file that has the unit tests, since that's eventually where the fuzz tests will live too. Wdyt?

Reply:
Makes sense, thanks for the suggestion!